common/scripted_effects/wc_artifacts_scripted_effects.txt

```diff
@@ -175,1100 +710,142 @@ give_generic_artifact_random_name_effect = {
-	}
-}
-
-spawn_artifact_amulet_2_intrigue_effect = {
-	add_artifact = amulet_2_intrigue
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_2_learning_effect = {
-	add_artifact = amulet_2_learning
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_3_diplomacy_effect = {
-	add_artifact = amulet_3_diplomacy
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_3_martial_effect = {
-	add_artifact = amulet_3_martial
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_3_stewardship_effect = {
-	add_artifact = amulet_3_stewardship
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_3_intrigue_effect = {
-	add_artifact = amulet_3_intrigue
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_3_learning_effect = {
-	add_artifact = amulet_3_learning
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_4_diplomacy_effect = {
-	add_artifact = amulet_4_diplomacy
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_4_martial_effect = {
-	add_artifact = amulet_4_martial
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_4_stewardship_effect = {
-	add_artifact = amulet_4_stewardship
-	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
-	}
-}
-
-spawn_artifact_amulet_4_intrigue_effect = {
-	add_artifact = amulet_4_intrigue
+give_new_artifact_random_name_and_origin_effect = {
 	new_artifact = {
-		give_generic_artifact_random_name_and_origin_effect = yes
-		finish_artifact_creation_effect = yes
+		if = {
+			limit = { quality == 1 }
+			set_description = "artifact_common_origin"
+		}
+		else_if = {
+			limit = { quality == 2 }
+			set_description = "artifact_uncommon_origin"
+		}
+		else_if = {
+			limit = { quality == 3 }
+			set_description = "artifact_rare_origin"
+		}
+		else_if = {
+			limit = { quality == 4 }
+			set_description = "artifact_epic_origin"
+		}
+		else = { set_description = "artifact_legendary_origin" }
+		give_artifact_random_name_effect = yes
+		set_creation_date = 1.1.1
+		log = "[This.OriginalOwner.GetTitledNameWithNick] has had [This.GetName] added to their Treasury."
```

Comment:
The log became harder to read. They are very different in length because of `[This.OriginalOwner.GetTitledNameWithNick]`, so artifact name is lost among other words.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/73669151-6d525980-46c0-11ea-8467-f606d5d4a5ab.png)

</details>

Reply:
Deleted, since it's no longer needed with current artifact spawn